src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -2214,15 +2215,9 @@ private static void RemoveReferencesWithoutConflicts
             Dictionary<string, List<AssemblyNameReference>> baseNameToReferences
         )
         {
-            string[] baseNames = new string[baseNameToReferences.Count];
-            baseNameToReferences.Keys.CopyTo(baseNames, 0);
-
-            foreach (string baseName in baseNames)
+            foreach (KeyValuePair<string, List<AssemblyNameReference>> kvp in baseNameToReferences.Where(baseName => baseName.Value.Count != 1))
```

Comment:
Could inline the check in the loop so a struct enumerator is used here and allocations are avoided.